rfcs/polyfill_testing.md

```diff
@@ -0,0 +1,151 @@
+  with future modifications.
+
+## Alternatives considered
+
+### Adding a canonical resource which includes the polyfill
+
+2. Add the polyfill into testharnessreport.js.
+3. Add the polyfill into a new /resources/polyfill.js which could also be
+   included by reftests.
+
+We could transparently insert the polyfill into a canonical resource which is
+included by most tests, for example `testharnessreport.js` or
+`/resources/polyfill.js`. Given that these external resources are expected to
+change it should not cause test failures due to the resource.
+
+Advantages:
+* Page structure and loaded resources are not modified.
+
+Concerns:
+* All tests which don't already include one of these resources (e.g. ref tests)
+  would need to be modified to include them. This would represent an ongoing
+  maintenance burden to ensure that new tests add these resources and failure to
+  do so wouldn't be caught by existing tests. Likely this would result in a
+  mixture of tests which support running with polyfills and tests which don't.
+* If we later adopt a mechanism which doesn't require explicitly including the
+  polyfill resources these additional artifacts may take some time to remove.
+
+### Bootstrap scripts feature
+
+Polyfill resources could be injected using the [bootstrap scripts feature](
+https://github.com/w3c/webdriver-bidi/issues/65). This would allow the polyfill
+to be transparently injected into tests.
+
+Advantages:
+* Polyfill is transparently inserted across all existing tests (including ref
+  tests).
+* Served files would match original test files.
+* As no modification is required, can easily switch mechanisms later.
+
+Concerns:
+* Developers would not be able to debug tests served from running `wpt serve` by
+  simply loading them in the browser.
+* This mechanism is not available yet.
+
+### Extension injection
+
+Similar to the bootstrap scripts feature, we could use an installed extension to
+inject the polyfill. As most polyfills would require installing functionality
+into the main script isolate, this would likely require injecting a script tag
+into the main document.
+
+Advantages:
+* Polyfill is transparently inserted across all existing tests (including ref
+  tests).
+* As no modification is required, can easily switch mechanisms later.
+
+Concerns:
+* As we would need to insert a script tag into the document, this would have the
+  same DOM observability concerns as the rewriting approach, with the same
+  possible mitigations.
+* Developers would not be able to debug tests served from running `wpt serve` by
+  simply loading them in the browser. They could however install the extension
+  which injected the polyfill.
+* Mobile browsers have no current support for these kinds of extensions.
+
+## Conclusion
+
+Rewriting the responses from the wpt server is a solution that is available now,
+supports all existing tests, and supports testing both via wpt run and by
```

Comment:
There are still some tests that don't work right, in particular XHTML and SVG? That seems fine, but worth spelling out still.

Reply:
Injecting on xhtml mime types seems straightforward enough, I added it to the polyfill. I've updated this to call out specifically html and xhtml tests.

Reply:
Cool, thanks!